### ISSUE TITLE
Mac app v2.0.3 doesn't launch from bundle, but does from CLI

### ISSUE THREAD

**Issue by incanus** [robhagemans/pcbasic#121]
First off, thank you _thank you_ @robhagemans for an amazing project. 

Congrats on the 2.0.3 release. 

Minor issue: I am seeing it not launch from Dock or directly from the app icon bundle, but something like `open /Applications/PC-BASIC.app` will work, as will going after the binary directly in `Contents/MacOS/pcbasic`. 

Here is the report from Console.app. I think it might be something related to the packaging process itself. 

```
default	10:31:00.293606-0700	loginwindow	-[PersistentAppsSupport applicationReady:] | App: PC-BASIC-2.0, ready, updating active tracking timer
default	10:31:00.293705-0700	loginwindow	-[ApplicationManager checkInAppContext:eventData:] | ApplicationManager: Checked in app : PC-BASIC-2.0
default	10:31:00.295269-0700	runningboardd	Resolved pid 24190 to [executable<pcbasic(501)>:24190]
default	10:31:00.296961-0700	runningboardd	[executable<pcbasic(501)>:24190] This process will not be managed.
default	10:31:00.297002-0700	runningboardd	Now tracking process: [executable<pcbasic(501)>:24190]
default	10:31:00.297583-0700	runningboardd	Acquiring assertion targeting executable<pcbasic(501)> from originator [daemon<com.apple.coreservices.launchservicesd>:256] with description <RBSAssertionDescriptor; foregroundApp:24190; ID: 476-256-102883; target: 24190> attributes = {
    <RBSDomainAttribute: 0x7fc7a4d50ed0; domain: com.apple.launchservicesd; name: RoleUserInteractiveNonFocal; sourceEnvironment: 0x0>;
}
default	2020-10-.29774902 10:31:00-0700	runningboardd	Assertion 476-256-102883 (target:executable<pcbasic(501)>) will be created as active
default	10:31:00.298351-0700	runningboardd	[executable<pcbasic(501)>:24190] Ignoring jetsam update because this process is not memory-managed
default	10:31:00.298549-0700	runningboardd	[executable<pcbasic(501)>:24190] Ignoring resume because this process is not lifecycle managed
default	10:31:00.298352-0700	runningboardd	Acquiring assertion targeting executable<pcbasic(501)> from originator [daemon<com.apple.coreservices.launchservicesd>:256] with description <RBSAssertionDescriptor; foregroundApp:24190; ID: 476-256-102884; target: 24190> attributes = {
    <RBSDomainAttribute: 0x7fc7a4d46630; domain: com.apple.launchservicesd; name: RoleUserInteractiveNonFocal; sourceEnvironment: 0x0>;
}
default	10:31:00.298706-0700	runningboardd	[executable<pcbasic(501)>:24190] Set darwin role to: UserInteractiveNonFocal
default	10:31:00.298866-0700	runningboardd	[executable<pcbasic(501)>:24190] Ignoring GPU update because this process is not GPU managed
default	10:31:00.298921-0700	runningboardd	Assertion 476-256-102884 (target:executable<pcbasic(501)>) will be created as active
default	10:31:00.299072-0700	runningboardd	Finished acquiring assertion 476-256-102883 (target:executable<pcbasic(501)>)
default	10:31:00.299902-0700	runningboardd	Invalidating assertion 476-256-102883 (target:executable<pcbasic(501)>) from originator 256
default	10:31:00.299909-0700	runningboardd	Finished acquiring assertion 476-256-102884 (target:executable<pcbasic(501)>)
default	10:31:00.441657-0700	runningboardd	[executable<pcbasic(501)>:24190] Death sentinel fired!
default	10:31:00.443545-0700	loginwindow	-[PersistentAppsSupport applicationQuit:] | for app:PC-BASIC-2.0, _appTrackingState = 2
default	10:31:00.443578-0700	loginwindow	-[PersistentAppsSupport applicationQuit:] | App: PC-BASIC-2.0, quit, updating active tracking timer
default	10:31:00.444787-0700	runningboardd	Invalidating assertion 476-256-102884 (target:executable<pcbasic(501)>) from originator 256
default	10:31:00.545583-0700	runningboardd	Removing process: [executable<pcbasic(501)>:24190]
default	10:31:00.556243-0700	runningboardd	Removing assertions for terminated process: [executable<pcbasic(501)>:24190]
```

Running the latest stable macOS 10.15.7. Also happens on a second machine running the latest Big Sur public beta 8 (`20A5374i`).

**Comment by robhagemans** [comment]
Thanks for the report - I'll have to look into it. Macs are weird... I thought I'd tried running it from the bundle, but maybe not. Also I'm working with an old version of MacOS. Given that ifrom the shell it seems to work, maybe there's some issue with it trying to write to standard i/o which I think doesn't exist when run from the dock; I've had issues with that on Mac before

**Comment by robhagemans** [comment]
I think this has been fixed with release 2.0.4 - it works for me now.
